packages/expo-dev-launcher/bundle/components/redesign/DeepLinkModal.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  View,
+  Heading,
+  Text,
+  Row,
+  XIcon,
+  Spacer,
+  Button,
+  StatusIndicator,
+  ChevronRightIcon,
+  Divider,
+} from 'expo-dev-client-components';
+import * as React from 'react';
+import { Alert, ScrollView } from 'react-native';
+
+import { useDevSessions } from '../../hooks/useDevSessions';
+import { useRecentlyOpenedApps } from '../../hooks/useRecentlyOpenedApps';
+import { loadApp } from '../../native-modules/DevLauncherInternal';
+
+export function DeepLinkModal({ onClosePress, pendingDeepLink }) {
```

Comment:
Could you split this component into smaller ones? Currently, it's hard to understand what going on there.